moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -350,8 +354,26 @@ void ompl_interface::ModelBasedPlanningContext::useConfig()
     }
     else
     {
-      objective =
-          std::make_shared<ompl::base::PathLengthOptimizationObjective>(ompl_simple_setup_->getSpaceInformation());
+        pluginlib::ClassLoader<ompl_optimization_loader::OptimizationObjectiveLoader> poly_loader("moveit_core", "ompl_optimization_loader::OptimizationObjectiveLoader");
+
+        try
+        {
+            RCLCPP_DEBUG(LOGGER, "Using optimization objective : %s", optimizer.c_str());
+            std::shared_ptr<ompl_optimization_loader::OptimizationObjectiveLoader> obj = poly_loader.createSharedInstance(optimizer);
```

Comment:
It depend on how the user creates his plugin_description.xml because the name tag allows the user to define the name he wants to use to load and therefor he can give a name consistent with the other objective, without any Loader.
If he doesn't give any name, yes he has to give the fully qualified type to the `optimization_objective` parameter. But because we have no choice to use a loader because of the parameter given to the constructor of the class OtpimizationObjective, I could not think to another solution.

Reply:
> the name tag allows the user to define the name he wants to use to load

Indeed. I guess I assumed everyone always uses the same name for C++ class name and plugin name.
I guess it is ok to deviate from that norm here.